pinot-controller/src/main/java/org/apache/pinot/controller/ControllerConf.java

```diff
@@ -146,6 +149,7 @@ private static long getRandomInitialDelayInSeconds() {
     private static final int DEFAULT_STATUS_CONTROLLER_WAIT_FOR_PUSH_TIME_IN_SECONDS = 10 * 60; // 10 minutes
     private static final int DEFAULT_TASK_MANAGER_FREQUENCY_IN_SECONDS = -1; // Disabled
     private static final int DEFAULT_MINION_INSTANCES_CLEANUP_TASK_FREQUENCY_IN_SECONDS = 60 * 60; // 1 Hour.
+    private static final int DEFAULT_MINION_INSTANCES_CLEANUP_TASK_MAX_OFFLINE_TIME_RANGE_IN_SECONDS = 10 * 60; // 10 minutes
```

Comment:
10 minutes might be too short, I'd suggest at least 1 hour

Reply:
make it configurable, default be 0 to keep current behavior

Reply:
Another scenario would be I just restarted the minion. IMO we should perform the following checks:
- Only cleanup non-live instance
- Only cleanup instances that has been offline for longer than this threshold

Reply:
Right, even for scenario when minion is just restarted, the diff between the current time and the value of `LAST_OFFLINE_TIME` won't be too long, i.e. the diff should be within this threshold.